### ISSUE TITLE
Idea: Bundle yarn with gdevelop - a great open source game engine 

### ISSUE THREAD

**Issue by blurymind** [InfiniteAmmoInc/Yarn#67]
Gdevelop is an incredibly fun and user friendly open source game engine that allows even people with no programming knowledge to create games.
https://gdevelop-app.com/
For programming it uses event sheets - similar to the ones in construct2 and clickteam fusion.
![events 88952ff1](https://user-images.githubusercontent.com/6495061/40881039-af294168-66b4-11e8-888c-e186ff390af5.png)

Now the reason for this proposal is gdevelop's new editor. I have been making some contributions to it lately and learning it's design - because it is written in javascript, using electron.

The editor is very user friendly, but it lacks storytelling tools - one has to make a story system in  it from scratch (even with event sheets).

Since both Yarn is web based and gdevelop's new editor is web based- I can try bundling yarn in it. But to do it, Yarn needs to be ported over to electron.
I already ported yarn to electron for a personal project myself- and got it to communicate with my app- so that is out of the way. I had to however change some of the code and add code to get it to work and open and save files like before.

I am thinking of using the excellent bondage.js for the story parsing part
https://www.npmjs.com/package/bondage
Many thanks to @jhayley for implementing it!

Posting here to see if there is community interested in this :)

Would you guys be interested to use yarn in an engine that is much smaller and easier to use than Unity3d? An engine that is both open source and actually would come packaged with yarn editor built into it? 
:)
related issue post: https://github.com/4ian/GD/issues/511

**Comment by charblar** [comment]
Sounds interesting, I believe this would be a question for @AlecHolowka to give his final say on.

**Comment by blurymind** [comment]
yes of course, also up to @4ian - Florian Rival - the creator of gdevelop.

I just personally feel that the two projects compliment each other on many levels - much more than yarn and Unity do.
Both Yarn and gdevelop are built on top of the same technology, they are both open source and target the same type of creative people- young story tellers/designers.

Because of the license, you can literaly bundle the two together and have them closely interact in ways that would be impossible with Unity.

The more I learn about how the two work, the more I see the potential of a collaboration of these two communities :)

Yarn and Gdevelop can be stronger together, because they have alot in common. They even look similar in terms of design. Wouldnt it be nice to be able to use yarn in a game engine out of the box- without having to rely on third party api/plugins. Have it open straight inside the game engine, and the game engine be really easy to set up the story
